common/clock/time_source.go

```diff
@@ -22,58 +22,44 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
+// Package timex provides extensions to the [time] package.
 package clock
 
 import (
-	"sync/atomic"
 	"time"
-
-	// clockwork is not currently used but it is useful to have the option to use this in testing code
-	// this comment is needed to stop lint from complaining about this _ import
-	_ "github.com/jonboulle/clockwork"
 )
 
 type (
-	// TimeSource is an interface for any
-	// entity that provides the current
-	// time. Its primarily used to mock
-	// out timesources in unit test
+	// TimeSource is an interface to make it easier to test code that uses time.
```

Comment:
I'm okay with adding stuff here slowly, but I guess I'd like to hear some plan for where this is going. We talked about the possibility of using clockwork directly or wrapping it. Now it looks like we're dropping it and going to reimplement everything? I thought you didn't want to go that way. What do we do if we need NewTimer, NewTicker, Sleep, etc.?